test/specs/components/action.spec.jsx

```diff
@@ -0,0 +1,541 @@
+import { mockComponent, findRenderedDOMComponentsWithId } from 'test/utils/react';
+
+import StripeCheckout from 'react-stripe-checkout';
+
+import { ActionComponent } from 'components/action.jsx';
+import { LoadingComponent } from 'components/loading.jsx';
+
+const stripeService = require('services/stripe-service');
+const userService = require('services/user-service');
+
+const AppStore = require('stores/app-store');
+
+const store = AppStore.store;
+
+function mockStore(s, state = {}) {
+    var mockedStore = getMockedStore(s, state);
+
+    Object.defineProperty(AppStore, 'store', {
+        get: () => {
+            return mockedStore;
+        }
+    });
+
+    return mockedStore;
+}
+
+const sandbox = sinon.sandbox.create();
+
+describe('Action', () => {
+    let component;
+    let componentNode;
+    let store;
+
+    beforeEach(() => {
+        mockComponent(sandbox, StripeCheckout, 'div', {
+            className: 'mockedStripe'
+        });
+        mockComponent(sandbox, LoadingComponent, 'div', {
+            className: 'mockedLoading'
+        });
+
+        sandbox.stub(userService, 'immediateUpdate');
+        userService.immediateUpdate.resolves();
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    after(() => {
+        Object.defineProperty(AppStore, 'store', {
+            get: () => {
+                return store;
+            }
+        });
+    });
+
+    describe('normal action link', () => {
+        const props = {
+            app: {},
+            text: 'action text',
+            uri: 'action uri'
+        };
+
+        beforeEach(() => {
+            component = TestUtils.renderIntoDocument(<ActionComponent {...props} />);
+            componentNode = ReactDOM.findDOMNode(component);
+        });
+
+        it('should render a link with target blank', () => {
+            const link = TestUtils.findRenderedDOMComponentWithTag(component, 'a');
+            link.textContent = 'action text';
+            link.href = 'action uri';
+            link.target = '_blank';
+        });
+    });
+
+    describe('javascript action link', () => {
+        const props = {
+            app: {},
+            text: 'action text',
+            uri: 'javascript:someAction()'
+        };
+
+        beforeEach(() => {
+            component = TestUtils.renderIntoDocument(<ActionComponent {...props} />);
+            componentNode = ReactDOM.findDOMNode(component);
+        });
+
+        it('should render a link with target self', () => {
+            const link = TestUtils.findRenderedDOMComponentWithTag(component, 'a');
+            link.textContent = 'action text';
+            link.href = 'javascript:someAction()';
+            link.target = '_self';
+        });
+    });
+
+    describe('buy action with stripe keys and offered state', () => {
+        const props = {
+            type: 'buy',
+            text: 'action text',
+            uri: 'fallback uri',
+            currency: 'USD',
```

Comment:
in fact it doesn't really matter, I do `this.props.currency.toUpperCase()` before passing it